git_hooks/set-up-git-hooks

```diff
@@ -11,14 +11,47 @@ for hook in "$hooks_dir"/*
 do
     file_name=$(basename "$hook")
     echo "\tCopying ${color_highlight}$file_name${color_none} hook to $git_hooks_dir and making it executable"
-    # if the pre-commit hook already exists, back up the existing one
-    if [ -f "$git_hooks_dir"/"$file_name" ]; then
-        backup_file_name="$file_name"-"$(date +%Y-%m-%d-%H-%M-%S)"
-        echo "\t\tbacking up existing $file_name as $backup_file_name"
-        mv "$git_hooks_dir"/"$file_name" "$git_hooks_dir"/"$backup_file_name"
+
+    if [ "$1" = "--overwrite" ]; then
+        echo "\t\toverwriting existing hook since ${color_highlight}--overwrite${color_none} option was used"
+    else
+        # if the pre-commit hook already exists, back up the existing one
+        if [ -f "$git_hooks_dir"/"$file_name" ]; then
+            backup_file_name="$file_name"-"$(date +%Y-%m-%d-%H-%M-%S)"
+            echo "\t\tbacking up existing $file_name as $backup_file_name"
+            mv "$git_hooks_dir"/"$file_name" "$git_hooks_dir"/"$backup_file_name"
+        fi
     fi
+
     cp "$hook" "$git_hooks_dir"
     chmod +x "$git_hooks_dir"/"$file_name"
 done
 
+echo "${color_highlight}Checking git hook dependencies...${color_none}"
+
+# check if detect-secrets is installed and prompt to install if it is not
+# if the user opts out of installing detect-secrets here, they will be prompted again
+# when they try to commit, assuming they have the secret scanning hook enabled
+if ! command -v detect-secrets &> /dev/null
+then
+    read -p $'\t\033[1;95mdetect-secrets\033[0m is not installed. Would you like to install it? (y/n): ' -r
+    if [[ $REPLY =~ ^[Yy]$ ]]
+    then
+        read -p $'\tWould you like to install using \033[1;95mpip\033[0m (p) or \033[1;95mbrew\033[0m (b)?: ' -r
```

Comment:
Didn't want to also use `{color_highlight}` and `{color_none}` here?

Reply:
I should've added a comment to explain, but I couldn't get the colour variables `{color_highlight}` and `{color_none}` to work with the `read` command using the `-p` prompt.

I could change these to use `echo` to display the prompt and then `read` in the input separately though

Reply:
Then it's probably not worth agonizing over.